### PR TITLE
[CI] Fix crashes of performance benchmark 

### PR DESCRIPTION
**Pull request by KuntaiDu**
The Llama 70B and mixtral benchmark is failing in the performance benchmark suite. This PR fixes it.
image

---

<details>

<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to vLLM! Before submitting the pull request, please ensure the PR meets the following criteria. This helps vLLM maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Only specific types of PRs will be reviewed. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Frontend]</code> For changes on the vLLM frontend (e.g., OpenAI API server, <code>LLM</code> class, etc.) </li>
    <li><code>[Kernel]</code> for changes affecting CUDA kernels or other compute kernels.</li>
    <li><code>[Core]</code> for changes in the core vLLM logic (e.g., <code>LLMEngine</code>, <code>AsyncLLMEngine</code>, <code>Scheduler</code>, etc.)</li>
    <li><code>[Hardware][Vendor]</code> for hardware-specific changes. Vendor name should appear in the prefix (e.g., <code>[Hardware][AMD]</code>).</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>We adhere to <a href="https://google.github.io/styleguide/pyguide.html">Google Python style guide</a> and <a href="https://google.github.io/styleguide/cppguide.html">Google C++ style guide</a>.</li>
    <li>Pass all linter checks. Please use <a href="https://github.com/vllm-project/vllm/blob/main/format.sh"><code>format.sh</code></a> to format your code.</li>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li>Include sufficient tests to ensure the project to stay correct and robust. This includes both unit tests and integration tests.</li>
    <li>Please add documentation to <code>docs/source/</code> if the PR modifies the user-facing behaviors of vLLM. It helps vLLM user understand and utilize the new features or changes.</li>
</ul>

<h3>Notes for Large Changes</h3>
<p>Please keep the changes as concise as possible. For major architectural changes (>500 LOC excluding kernel/data/config/test), we would expect a GitHub issue (RFC) discussing the technical design and justification. Otherwise, we will tag it with <code>rfc-required</code> and might not go through the PR.</p>

<h3>What to Expect for the Reviews</h3>

<p>The goal of the vLLM team is to be a <i>transparent reviewing machine</i>. We would like to make the review process transparent and efficient and make sure no contributor feel confused or frustrated. However, the vLLM team is small, so we need to prioritize some PRs over others. Here is what you can expect from the review process: </p>

<ul>
    <li> After the PR is submitted, the PR will be assigned to a reviewer. Every reviewer will pick up the PRs based on their expertise and availability.</li>
    <li> After the PR is assigned, the reviewer will provide status update every 2-3 days. If the PR is not reviewed within 7 days, please feel free to ping the reviewer or the vLLM team.</li>
    <li> After the review, the reviewer will put an <code> action-required</code> label on the PR if there are changes required. The contributor should address the comments and ping the reviewer to re-review the PR.</li>
    <li> Please respond to all comments within a reasonable time frame. If a comment isn't clear or you disagree with a suggestion, feel free to ask for clarification or discuss the suggestion.
 </li>
</ul>

<h3>Thank You</h3>

<p> Finally, thank you for taking the time to read these guidelines and for your interest in contributing to vLLM. Your contributions make vLLM a great tool for everyone! </p>

</details>

### CODE REVIEW

**Comment by KuntaiDu** [comment]
Don't merge this PR for now... This PR is not ready, and I add `/ready` tag only to trigger aws CI, which is a dependency for performance benchmark.

**Comment by KuntaiDu** [comment]
Now this PR is ready to merge --- all serving benchmark results appeared, including Llama 70B and Mixtral 8x7B.
https://buildkite.com/vllm/performance-benchmark/builds/8182

File: .buildkite/nightly-benchmarks/run-benchmarks-suite.sh [hunk 1]
```diff
@@ -70,24 +70,11 @@ wait_for_server() {
...
-  else
-      for pid in $pids; do
-          kill -9 "$pid"
-          echo "Killed process with PID: $pid"
-      done
 
-      echo "All GPU processes have been killed."
-  fi
+  pkill -f python3
+  ps -e | grep pt_main_thread | awk '{print $1}' | xargs kill -9
```
Comment (simon-mo) [inline .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:75]: This was removed by @cadedaniel

Comment (cadedaniel) [reply .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:75]: the `pkill -f python3` makes it hard to run this locally when other python processes live. happy to help craft a better command.

Comment (cadedaniel) [reply .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:75]: optionally, can have a flag to skip this kill if provided. im happy to make a PR for that

Comment (KuntaiDu) [reply .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:75]: Currently the command `pkill-f python3` is mainly used to kill the openAI server -- it is now in a separate process. Would be great if @cadedaniel you know a graceful way to shutdown the openAI server.

Comment (cadedaniel) [reply .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:75]: `ps aux | grep python | grep openai | awk '{print $2}' | xargs -r kill -9`

**Comment by KuntaiDu** [comment]
Fixed and the new performance benchmark runs normally.
https://buildkite.com/vllm/performance-benchmark/builds/8188

File: .buildkite/nightly-benchmarks/run-benchmarks-suite.sh [hunk 2]
```diff
@@ -70,24 +70,11 @@ wait_for_server() {
...
+  ps aux | grep python | grep openai | awk '{print $2}' | xargs -r kill -9
+  ps -e | grep pt_main_thread | awk '{print $1}' | xargs kill -9
 
-  # waiting for GPU processes to be fully killed
-  # loop while nvidia-smi returns any processes
-  while [ -n "$(nvidia-smi --query-compute-apps=pid --format=csv,noheader)" ]; do
-    sleep 1
-    echo "Waiting for GPU processes to be killed"
-  done
+  sleep 5
```
Comment (cadedaniel) [inline .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:77]: nit: better to wait until GPU memory is free to reduce flakiness (can use https://github.com/vllm-project/vllm/blob/3f674a49b5033a6ed778ab960e86e03cfa64aa1f/tests/utils.py#L332)

**Review by cadedaniel — APPROVED** [review]

File: .buildkite/nightly-benchmarks/run-benchmarks-suite.sh (hunk 2 above)
Comment (KuntaiDu) [reply .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:77]: Sure let me fix that

Comment (KuntaiDu) [reply .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:77]: BTW is it OK for me to bump up the Llama version to 3.1 for specdecode test here 
```
    {
        "test_name": "serving_llama70B_tp4_sharegpt_specdecode",
        "qps_list": [2],
        "server_parameters": {
            "model": "meta-llama/Meta-Llama-3-70B-Instruct",
            "disable_log_requests": "", 
            "tensor_parallel_size": 4,
            "swap_space": 16, 
            "speculative_model": "turboderp/Qwama-0.5B-Instruct",
            "num_speculative_tokens": 4,
            "speculative_draft_tensor_parallel_size": 1,
            "use_v2_block_manager": ""
        },
        "client_parameters": {
            "model": "meta-llama/Meta-Llama-3-70B-Instruct",
            "backend": "vllm",
            "dataset_name": "sharegpt",
            "dataset_path": "./ShareGPT_V3_unfiltered_cleaned_split.json",
            "num_prompts": 200 
        }
    }
```

Comment (cadedaniel) [reply .buildkite/nightly-benchmarks/run-benchmarks-suite.sh:77]: sure

**Comment by KuntaiDu** [comment]
Comments fixed, ready for merge if the performance benchmark returns normally.

**Review by simon-mo — APPROVED** [review]

**Pull request merged**